src/serialize.rs

```diff
@@ -672,6 +710,83 @@ impl Serialize for [u8] {
     }
 }
 
+
+#[cfg(feature = "arraystring")]
+pub struct ArrayStringReader<const CAP: usize> {
+    bytes_reader: BytesReader,
+}
+
+#[cfg(feature = "arraystring")]
+impl<const CAP: usize> TypeRead for ArrayStringReader<CAP> {
+    type Value = ArrayString<CAP>;
+
+    fn read_one<R: io::Read>(&self, bytes: R) -> io::Result<Self::Value>
+    {
+        let v = self.bytes_reader.read_one(bytes)?;
+        let mut data = [0_u8; CAP];
+        v.iter().take(CAP).enumerate().for_each(|(i, c)| {
+            data[i] = *c;
+        });
+        match ArrayString::<CAP>::from_byte_string(&data) {
+            Ok(s) => Ok(s),
+            Err(e) => Err(io::Error::new(
+                std::io::ErrorKind::InvalidData,
+                format!("Can not convert to utf-8: {}", e),
+            )),
+        }
+    }
+}
+
+#[cfg(feature = "arraystring")]
+impl<const CAP: usize> Deserialize for ArrayString<CAP> {
+    type TypeReader = ArrayStringReader<CAP>;
+
+    fn reader(dtype: &DType) -> Result<Self::TypeReader, DTypeError> {
+        Ok(ArrayStringReader::<CAP> {
+            bytes_reader: <Vec<u8> as Deserialize>::reader(dtype)?,
+        })
+    }
+}
+
+#[cfg(feature = "arraystring")]
+pub struct ArrayStringWriter<const CAP: usize> {
```

Comment:
Hmm, I'm not sure why this should be necessary?  I would think that both ArrayString and String should be able to use a single TypeWrite for writing `|V` strings.  Something like:

```rust
pub struct Utf8Writer {
    type Value = str;
    ....
}
```